### ISSUE TITLE
Incorrect rendering if the size of the screen is bigger than the size of the map

### ISSUE THREAD

**Issue by abacabadabacaba** [Globulation2/glob2#52]
Globulation 2 - 0.9.4.4

If the screen is bigger than the map, then some parts of the map can appear on the screen multiple times. When this happens, the terrain is rendered correctly, but units and buildings are only drawn once.

Everything should be drawn as many times as it appears on the screen: terrain, units, buildings, target lines, etc.

**Comment by stephanemagnenat** [comment]
Yes, this is a known bug, it happened because when Glob2 was originally developed, the screens were much smaller ;-)

**Comment by stephanemagnenat** [comment]
This happens because as buildings and units can span multiple tiles on the map, they are first put to a set, and then drawn once. A fix to this issue would be to check whether the screen is bigger than the whole map, and then draw multiple times. 

The locations to implement the corrections can be found from `Game::drawMap` in `src/Game.cpp:2807`.

**Comment by stephanemagnenat** [comment]
PR #88 is a first step towards solving this issue.
